src/coreclr/tools/aot/ILLink.Shared/DiagnosticId.cs

```diff
@@ -210,6 +209,8 @@ public static class DiagnosticIdExtensions
                 2106 => MessageSubCategory.TrimAnalysis,
                 2107 => MessageSubCategory.TrimAnalysis,
                 var x when x >= 2109 && x <= 2116 => MessageSubCategory.TrimAnalysis,
+                var x when x >= 3050 && x <= 3052 => MessageSubCategory.AotAnalysis,
```

Comment:
I guess this matches the existing style, but the C# way of doing this (that also generates more efficient code) is:

```suggestion
                >= 3050 and <= 3052 => MessageSubCategory.AotAnalysis,
```

Reply:
I will update Linker with the changes in resources and DiagnosticIds from NativeAOT, so I will include this change in that PR.